low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -32,18 +44,125 @@ class ServerMap(ResourceMap):
 
     def __init__(self):
         """Initialize server"""
-        pass
+        self.server_map = {
+            "memory": self.get_mem_info
+            }
 
     def get_health_info(self, rpath):
         """
         Get health information of fru in given resource id
 
         rpath: Resource id (Example: nodes[0]>compute[0]>hw>disks)
         """
-        pass
+        info = {}
+        nodes = rpath.strip().split(">")
+        leaf_node, _ = self.get_node_details(nodes[-1])
+        if leaf_node == "compute":
+            info["last_updated"] = int(time.time())
+            for fru in self.server_map:
+                info.update({fru: self.server_map[fru]()})
+        else:
+            fru = None
+            fru_found = False
+            for node in nodes:
+                fru, _ = self.get_node_details(node)
+                if self.server_map.get(fru):
+                    fru_found = True
+                    info = self.server_map[fru]()
+                    break
+        if not fru_found:
+            raise ResourceMapError(
+                errno.EINVAL,
+                f"Health provider doesn't have support for'{rpath}'.")
+        return info
+
+    def get_mem_info(self):
+        """Update and return memory info in specific format"""
+
+        from framework.utils.store_factory import file_store
+        node_id = Conf.get(GLOBAL_CONF, NODE_ID_KEY,'SN01')
+        self.store = file_store
+        cache_dir_path = os.path.join(DATA_PATH, 'server')
+        self.MEM_FAULT_SENSOR_DATA = os.path.join(cache_dir_path,
+                                             f'MEM_FAULT_SENSOR_DATA_{node_id}')
+        data = []
+        status = "OK"
+        description = "Host memory is in good health"
+        utility_instance = ToolFactory().get_instance('procfs')
+        mem_path = utility_instance.get_proc_memory('meminfo')
+        if mem_path.is_file():
+            meminfo_dict = self.read_mem_file(mem_path)
+            total_mem = meminfo_dict['MemTotal:']
+            path_exists, _ = self.store.exists(self.MEM_FAULT_SENSOR_DATA)
+            if path_exists:
+                self.get_stored_mem_info()
+                if self.fault_alert_state == "fault":
```

Comment:
As discussed, we dont see any way to detect DIMM fault in sensor also. All it does is to check at SSPL startup, about what was last seen total system memory( it would be based on available DIMMs) and compare with current memory value & raise alerts.
This doesnt give idea about current memory issue, if any. So wrt health, we will give current system memory usage and if its crossing configured threshold value

Reply:
Ack.
Done the changes for the same